tgui/packages/tgui/interfaces/Manufacturer/index.tsx

```diff
@@ -83,26 +83,30 @@ export const Manufacturer = (_, context) => {
   }
 
   // Get a ManufacturableData from a QueueBlueprint using its type, category, and name.
-  let queueBlueprintRefs = data.queue.map((queued:QueueBlueprint) =>
+  let queueBlueprintRefs = data.queue && data.queue.length
+  && data.queue.map((queued:QueueBlueprint) =>
     blueprints_by_category[queued.category].find((key) => (key.name === queued.name))
   );
 
   return (
     <Window width={1200} height={600} title={data.fabricator_name}>
-      {!data.indicators.hasPower && <PowerAlertModal width={100-SETTINGS_WINDOW_WIDTH} height={"100%"} />}
+      {data.indicators && !data.indicators.hasPower
```

Comment:
Confirming intended behavior: if `data.indicators` is not set, as you have it this will not show the `PowerAlertModal`.

Further down I think you treat `data.indicators` not being set as out of power?

Broadly, rather than burrowing down each time, just extract it once then use throughout (i.e. here, and wherever else you care about it)?

```
const hasPower = !!data.indicators?.hasPower;
```

Reply:
Sounds good! Did that, and it fixed one of the errant null errors too.